include/picongpu/fields/laserProfiles/laserExpRampWithPrepulse.hpp

```diff
@@ -0,0 +1,198 @@
+                AMP_PREPULSE * gauss( runTime - TIME_PREPULSE );
+        }
+        else if ( before_peakpulse )
+        {
+            const float_X ramp_when_peakpulse = extrapolate_expo(
+                TIME_2,
+                AMP_2,
+                TIME_3,
+                AMP_3,
+                endUpramp
+            ) / AMPLITUDE;
+            // if (ramp_when_peakpulse > 0.5) - I know, dead code :) didn't understand your comment about the other logging
+            //    throw std::invalid_argument("\n\nAttention, the intensities of the ramp are very large, the extrapolation to the time of the main pulse would give more than 50% of the pulse amplitude - this is not a gaussian pulse at all anymore, probably something wrong?!\n");
+
+            env += AMPLITUDE * ( 1. - ramp_when_peakpulse ) *
+                gauss( runTime - endUpramp );
+            env += AMP_PREPULSE * gauss( runTime - TIME_PREPULSE );
+            if ( during_first_exp )
+                env += extrapolate_expo(
+                    TIME_1,
+                    AMP_1,
+                    TIME_2,
+                    AMP_2,
+                    runTime
+                );
+            else
+                env += extrapolate_expo(
+                    TIME_2,
+                    AMP_2,
+                    TIME_3,
+                    AMP_3,
+                    runTime
+                    );
+        }
+        else if ( ! after_peakpulse )
+            env = AMPLITUDE;
+    else // after startDownramp
+            env = AMPLITUDE * gauss( runTime - startDownramp );
+        return env;
+    }
+
+    HINLINE float3_X laserLongitudinal(uint32_t currentStep, float_X& phase)
+    {
+        float_X envelope;
+        float3_X elong( float3_X::create( 0.0 ) );
+
+        // a symmetric pulse will be initialized at position z=0 for
+        // a time of RAMP_INIT * PULSE_LENGTH + LASER_NOFOCUS_CONSTANT = INIT_TIME.
+        // we shift the complete pulse for the half of this time to start with
+        // the front of the laser pulse.
+
+        /* initialize the laser not in the first cell is equal to a negative shift
+         * in time
+         */
+        const float_64 runTime = (DELTA_T * currentStep - laserTimeShift -
+            0.5 * RAMP_INIT * PULSE_LENGTH );
+
+        phase += float_X( w * runTime ) + LASER_PHASE ;
+
+        envelope = get_envelope( runTime );
+
+
+        if( Polarisation == LINEAR_X )
+        {
+            elong.x() = float_X( envelope * ( math::sin( phase ) ) );
+        }
+        else if( Polarisation == LINEAR_Z )
+        {
+            elong.z() = float_X( envelope * ( math::sin( phase ) ) );
+        }
+        else if( Polarisation == CIRCULAR )
+        {
+            elong.x() = float_X( envelope / sqrt( 2.0 ) * ( math::sin( phase ) ) );
+            elong.z() = float_X( envelope / sqrt( 2.0 ) * ( math::cos( phase ) ) );
+        }
+
+        return elong;
+    }
+
+    /**
+     *
+     * @param elong
```

Comment:
doxygen descriptions missing

Reply:
this is the same in all profiles; well, I've tried to explain it in a few words